### ISSUE TITLE
Anticipation status of exogenized variables

### ISSUE THREAD

**Issue by PatMcFarlane** [IRIS-Solutions-Team/IRIS-Toolbox#247]
This is not an issue...just a question.

I'm trying to exogenize some variables, beginning from the 3rd period of my simulation.  

For some of these exogenized variables, I would like the shocks used to achieve the desired path to be unanticipated.  The ideal setup would be if they were unanticipated up to period 3 but then anticipated from that point forward.  If that is not possible, I would settle for the whole sequence of shocks being unanticipated.

I tried using P=endogenize(P,startDate+2:endDate,'ShockName','Anticipate=',false) but that did not work.  

Thanks in advance,

Patrick

**Comment by PatMcFarlane** [comment]
I think I figured out how to make the whole sequence unanticipated:

```
p=Plan(M,startDate:endDate);
p=anticipate(p,false,{'var_exogenized','var_endogenized'});
p=exogenize(p,startDate+2:endDate,'var_exogenized');
p=endogenize(p,startDate+2:endDate,'var_endogenize');
```

**Comment by PatMcFarlane** [comment]
I think I've also figured out how to set it up to have the exogenized paths remain unanticipated until the endogenized shocks kick-in:

```
leadup=3; %number of periods before exogenizing (exog path is unanticipated during these)
d=sstatedb(M,startDate:endDate);
s1=simulate(M,d,startDate:endDate);
d2=s1;
p=plan(M,startDate + leadup - 1 : endDate);
p=exogenize(p,startDate + leadup - 1: EndExog,'Y');
p=endogenize(p,startDate + leadup - 1: EndExog,'eY');
s2=simulate(M,d2,startDate + leadup -1:endDate),'Plan=',p);
```

**Comment by jaromir-benes** [comment]
Yes, indeed. I was just about the respond with the solution as well. You
can always manually split the information set into whatever you need. We
have plans for more complex information sets within the Plan object - just
a feature of convenience.

Also note, for computational efficiency, you don't need to run the first
simulation all the way to endDate - you can stop one period before the
start of the next simulation. Also, I recommend using the option
PrependInput=true to combine the output time series with the presample
values from the input databank.

Best
Jaromir
